### DISCUSSION TITLE
suggested Style Guide for each chapter's Top Recommendations for Site Owners

### DISCUSSION THREAD

**Discussion by cqueern** [HTTPArchive/almanac.httparchive.org#2178] [Ideas]
Making the Web Almanac actionable can accelerate a better web for all.  Here's a suggested style guide for a section called **Top Recommendations for Site Owners** that Web Almanac authors can use to conclude future chapters of the Web Almanac in a consistent experience for Web Almanac readers.

## Guidelines
 * Chapter authors should share **between one and three recommendations** that site owners can implement on their site that improve the web. More than three is discouraged.
* Each recommendation should be composed of:
  * **one summary statement** that conveys the recommendation in 256 characters (including spaces) or less (why 256? This allows the [HTTPArchive twitter account](https://twitter.com/httparchive) to tweet your recommendation in your words and still has [23 characters for the link](https://developer.twitter.com/en/docs/counting-characters) back to your Archive chapter and one space between your recommendation and the link.)
  * **one paragraph** of no more than 50 words of further explanation
* Ideal recommendations deliver **high value with low level of effort**. Not all recommendations will meet this criteria but if we aim for that and hit it most of the time, we'll be helping a lot of people. 
* Recommendations should be **tied to analysis shared earlier in the chapter**. This is not a place to introduce new concepts.

## Visual Example

The Web Almanac's design team will of course have thoughts on presenting the recommendations in terms of spacing, font size and style, etc. The example below gives an idea of visual ratio of summary statements to their corresponding explanations given the above character and word limits.

#### Top Recommendations for Site Owners
**1. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam molestie pellentesque mauris, ut maximus eget. Phasellus mauris, mattis dolor eget.**
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce in nibh non est tincidunt semper vitae sed urna. Duis lacinia, nulla at posuere dapibus, magna leo tincidunt nulla, nec consequat ante lorem cursus nulla. Proin mauris dolor, elementum et facilisis ac, pulvinar eu tortor. Suspendisse vel vulputate ligula, sed tristique.

**2. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam molestie pellentesque mauris, ut maximus eget. Phasellus mauris, mattis dolor eget.**
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce in nibh non est tincidunt semper vitae sed urna. Duis lacinia, nulla at posuere dapibus, magna leo tincidunt nulla, nec consequat ante lorem cursus nulla. Proin mauris dolor, elementum et facilisis ac, pulvinar eu tortor. Suspendisse vel vulputate ligula, sed tristique.

**3. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam molestie pellentesque mauris, ut maximus eget. Phasellus mauris, mattis dolor eget.**
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce in nibh non est tincidunt semper vitae sed urna. Duis lacinia, nulla at posuere dapibus, magna leo tincidunt nulla, nec consequat ante lorem cursus nulla. Proin mauris dolor, elementum et facilisis ac, pulvinar eu tortor. Suspendisse vel vulputate ligula, sed tristique.

Thanks for reading.  I look forward to feedback.

**Comment by roryhewitt** [comment]
Might want three separate bits per recommendation:

1. Highlight - no more that 10 words, not necessarily a full sentence - designed to be a 'punchy' attention-getter
2. Title - 30-40 words, should be a proper sentence, as described above
3. Description - as described above

That way, the **Highlight** can be displayed in a format designed to catch people's attention, like we have the three highlights from each chapter already - perhaps when someone is browsing the main page, it could show something like "three highlights from the *Caching* chapter and one recommendation from the *Third-Parties* chapter"?

Here's a dummy example I would have added from the 2020 Caching chapter:

**IMPLEMENT CACHE REVALIDATION!**

**Add ETag and/or Last-Modified support, to allow for cache revalidation and reduced bandwidth and latency**

*Adding support at the server for the ETag response header (or the Last-Modified response header if ETag support is not possible) can result in significant bandwidth and latency reduction due to more 304 Not Modified responses being returned and allowing browser-cached responses to be used more effectively*

**Comment by cqueern** [comment]
@rviscomi  @tunetheweb hi guys. Any thoughts on...

- deciding on a format for chapter authors to use for Top Recommendations
- communicating it to chapter authors

...? 

Thanks!

**Reply by rviscomi** [reply]
@cqueern let's mock up a prototype using an actual chapter to get a better sense of how it'd look and feel and iterate on it from there. Would you be up for creating a new branch to build and test this out?

**Reply by cqueern** [reply]
Hey Guys. Sure. I can take a shot. I may have questions but I can reach out on slack for those. Will likely be this weekend or early next week.

**Reply by cqueern** [reply]
> @cqueern let's mock up a prototype using an actual chapter to get a better sense of how it'd look and feel and iterate on it from there. Would you be up for creating a new branch to build and test this out?

Team I apologize - I'm not likely to get to this in the near future given current obligations. @rviscomi @tunetheweb might this request be reassigned to someone if it's still important?

I do still think each chapter's top recommendations for site owners would be a valuable enhancement for the audience so I hope it can make it in this year's edition somehow...

**Reply by rviscomi** [reply]
I'd hate to drop the ball but I'm not sure if I have the time to commit to this either. Open to letting anyone else interested take a crack at it.
